crates/dt-workspace/src/lib.rs

```
@@ -0,0 +1,58 @@
+use std::{
+    borrow::Cow,
+    path::{Path, PathBuf},
+};
+
+use config::{Config, ConfigError};
+use thiserror::Error;
+
+pub mod config;
+mod search;
+
+#[derive(Debug)]
+pub struct Workspace {
+    path: PathBuf,
+
+    config: Config,
+}
+
+impl Workspace {
+    pub fn try_new(override_path: Cow<'static, Path>) -> Result<Self, WorkspaceError> {
+        let mut path = override_path.to_path_buf();
+
+        if !path.is_dir() {
+            path = PathBuf::from(".");
+        }
```

```suggestion
```
This shouldn't silently use cwd. The caller will validate the path if needed and Config::load_from_parent will error. See what I'll write in a moment: